components/Toast/index.tsx

```diff
@@ -0,0 +1,33 @@
+'use client';
+
+import { toast } from 'react-toastify';
+import { StyledToast, UndoButton } from './style';
+
+interface ToastOptions {
+  message: string;
+  undo?: boolean;
+  undoAction?: () => void;
+}
+
+export function showToast({ message, undo, undoAction }: ToastOptions) {
+  toast(
+    ({ closeToast }) => (
+      <StyledToast>
```

Comment:
<img width="369" alt="Screenshot 2025-04-27 at 11 42 48 AM" src="https://github.com/user-attachments/assets/351eff8e-1468-4849-a306-80bed3b0046a" />
The toast progress bar has a little space on the bottom left since its curved